Also match lld's "compatible with GNU linkers".

Probably not really used at current (1.1.1a) as $config{CC} isn't even
set up when this is called...

Index: Configurations/shared-info.pl
--- Configurations/shared-info.pl.orig
+++ Configurations/shared-info.pl
@@ -14,7 +14,7 @@
 sub detect_gnu_ld {
     my @lines =
         `$config{CROSS_COMPILE}$config{CC} -Wl,-V /dev/null 2>&1`;
-    return grep /^GNU ld/, @lines;
+    return grep /^GNU ld|GNU linker/, @lines;
 }
 sub detect_gnu_cc {
     my @lines =
